sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -358,37 +289,39 @@ def mark_upstream_ios_of_output(template_output, marked_inputs,
 
     # Converting pipeline inputs
     pipeline, pipeline_workspaces = big_data_passing_pipeline(
-        pipeline_template, inputs_consumed_as_artifacts,
+        pipelinerun_name, pipeline_template, inputs_consumed_as_artifacts,
         output_tasks_consumed_as_artifacts)
 
     # Add workspaces to pipelinerun if big data passing
-    # Check whether pipelinerun was generated, throw error if not.
     if pipeline_workspaces:
         pipeline, pipelinerun_workspaces = big_data_passing_pipelinerun(
-            pipelinerun_template, pipeline_workspaces)
+            pipelinerun_name, pipelinerun_template, pipeline_workspaces)
 
     # Use workspaces to tasks if big data passing instead of 'results', 'copy-inputs'
     for task_template in container_templates:
         task_template = big_data_passing_tasks(task_template,
                                                inputs_consumed_as_artifacts,
                                                outputs_consumed_as_artifacts)
 
-    # Create pvc for pipelinerun if big data passing.
-    # As we used workspaces in tekton pipelines which depends on it.
-    # User need to create PV manually, or enable dynamic volume provisioning, refer to the link of:
-    # https://kubernetes.io/docs/concepts/storage/dynamic-provisioning
-    # TODO: Remove PVC if Tekton version > = 0.12, use 'volumeClaimTemplate' instead
-    if pipelinerun_workspaces:
-        for pipelinerun in pipelinerun_workspaces:
-            workflow.append(create_pvc(pipelinerun))
+    # # Create pvc for pipelinerun if big data passing.
+    # # As we used workspaces in tekton pipelines which depends on it.
+    # # User need to create PV manually, or enable dynamic volume provisioning, refer to the link of:
+    # # https://kubernetes.io/docs/concepts/storage/dynamic-provisioning
+    # # TODO: Remove PVC, use 'volumeClaimTemplate' instead, issue #181.
+    # User need to create pvc manually recently until issue #181 addressed
```

Comment:
Maybe we can put these comment in the sdk readme? This way users can be aware of the big_data_passing prerequisites

Reply:
addressed